File: Solved/p00299.cpp
/*
*
* Problema UVA 299 - Train Swapping
*
*/

#include <stdio.h>
#include <iostream>

int main(){
	int cases;
	int length;
	int train[50];
	int i, j, aux;
	int swaps, moves;

	scanf("%d\n", &cases);

	while(cases--){
		scanf("%d\n", &length);
		for(i = 0; i < length; i++){
			scanf("%d", &train[i]);
		}

		swaps = 0;

		for(i = 1 ; i < length; i++){
			aux = train[i];
			j = i-1;
			moves = 0;
			while( j >= 0 && train[j] > aux ){
				train[j+1] = train[j];
				j--;
				moves++;
				swaps++;
			}
			train[j+1] = aux;
		}

		printf("Optimal train swapping takes %d swaps.\n", swaps);
	}


	return 0;
}

File: Solved/p00299.py
'''
*
* Problema UVa 299 - Train Swapping
*
'''

from sys import *
##import time

def sort(lst):
	count = 0

	i = 0
	while i < len(lst)-1:
		a = lst[i]
		b = lst[i+1]
		if a > b:
			lst[i], lst[i+1] = lst[i+1], lst[i]
			i = 0
			count += 1
		else:
			i += 1
	return count

def main():
	lines = stdin.readlines()
	lines.pop(0)
	out = ""
	for i, line in enumerate(lines):
		if (i+1)%2 == 0: 
			count = 0
			line = line.strip().split()
			line = list(map(int, line))
			out += "Optimal train swapping takes {} swaps.\n".format(sort(line))
	print(out.strip())

##startTime = time.time()
main()
##print("--- %s seconds ---" % (time.time() - startTime))